Detectors/Calibration/src/MeanVertexCalibrator.cxx

```diff
@@ -35,137 +36,289 @@ void MeanVertexCalibrator::initOutput()
-    if (fitres >= 0) {
-      LOG(info) << "Y: Fit result (of merged Slots) => " << fitres << ". Mean = " << fitValues[1] << " Sigma = " << fitValues[2];
-    } else {
-      LOG(warning) << "Y: Fit failed with result = " << fitres;
+      LOG(info) << "Adding point " << i << ": zvtx = " << meanZvect[i] << " xvtx = " << fitResSlicesX[i][2];
     }
-    mSMAMVobj.setY(fitValues[1]);
-    mSMAMVobj.setSigmaY(fitValues[2]);
+    lf.AddPoint(&meanZvect[i], fitResSlicesX[i][1]);
+  }
+  lf.Eval();
+  double slopeX = lf.GetParameter(1);
+  mvo.setSlopeX(slopeX);
+  mvo.setX(mvo.getZ() * slopeX + lf.GetParameter(0));
+  lf.ClearPoints();
 
-    // z coordinate
+  // now slope for the y-coordinate dependence on z
+  for (int i = 0; i < fitResSlicesX.size(); ++i) {
     if (mVerbose) {
-      LOG(info) << "**** Printing content of SMA MVData object for z coordinate";
-      for (int i = 0; i < mSMAdata.nbinsZ; i++) {
-        LOG(info) << "i = " << i << ", content of histogram = " << mSMAdata.histoZ[i];
-      }
+      LOG(info) << "Adding point " << i << ": zvtx = " << meanZvect[i] << " yvtx = " << fitResSlicesY[i][2];
     }
-    array = &mSMAdata.histoZ[0];
-    fitres = fitGaus(mSMAdata.nbinsZ, array, -(mSMAdata.rangeZ), mSMAdata.rangeZ, fitValues);
-    if (fitres >= 0) {
-      LOG(info) << "Z: Fit result (of merged Slots) => " << fitres << ". Mean = " << fitValues[1] << " Sigma = " << fitValues[2];
+    lf.AddPoint(&meanZvect[i], fitResSlicesY[i][1]);
+  }
+  lf.Eval();
+  double slopeY = lf.GetParameter(1);
+  mvo.setSlopeY(slopeY);
+  mvo.setY(mvo.getZ() * slopeY + lf.GetParameter(0));
+  if (mVerbose) {
+    LOG(info) << "slope X = " << slopeX;
+    LOG(info) << "slope Y = " << slopeY;
+  }
+}
+//_____________________________________________
+void MeanVertexCalibrator::fitMeanVertexCoord(int icoord, int nbins, float* array, float minRange, float maxRange, MeanVertexObject& mvo)
+{
+  // fit mean vertex coordinate icoord
+  std::vector<float> fitValues;
+  float binWidth = 0;
+  if (mVerbose) {
+    LOG(info) << "**** Printing content of MeanVertex object for coordinate " << icoord;
+    if (icoord == 0) {
+      binWidth = mBinWidthX;
+    } else if (icoord == 1) {
+      binWidth = mBinWidthY;
     } else {
-      LOG(warning) << "Z: Fit failed with result = " << fitres;
+      binWidth = mBinWidthZ;
     }
-    mSMAMVobj.setZ(fitValues[1]);
-    mSMAMVobj.setSigmaZ(fitValues[2]);
+    printVector(array, nbins, minRange, maxRange, binWidth);
   }
+  double fitres = fitGaus(nbins, array, minRange, maxRange, fitValues);
+  if (fitres != -4) {
+    LOG(info) << "coordinate " << icoord << ": Fit result of full statistics => " << fitres << ". Mean = " << fitValues[1] << " Sigma = " << fitValues[2];
+  } else {
+    LOG(error) << "coordinate " << icoord << ": Fit failed with result = " << fitres;
+  }
+  switch (icoord) {
+    case 0:
+      mvo.setX(fitValues[1]);
+      mvo.setSigmaX(fitValues[2]);
```

Comment:
you are missing `break;`.

Reply:
What a stupidity... Sorry. I did not compile because I am working on a different branch, and wanted to do this fast.

Reply:
Note that I did not add a default, as these are the only options, and default should not be there.